pick.lic

```diff
@@ -360,6 +363,10 @@ class Pick
         end
 
         while holding_box?(current_box) && current_box['trapped'] && !current_box['trap_difficulty'].nil?
+          if current_box['trap'] && @trap_greylist.include?(current_box['trap'])
+            current_box['trap_difficulty'] = 18
+            #18 is one above the player range to block potential yaml setting interference
+          end        
```

Comment:
As I was trying to understand why there was a loop within a loop disarming a box, I remembered that boxes can have multiple traps to disarm. Got it, so we're iterating over each possible trap. So I now understand why your greylist check is within this loop so that even if the 1st trap wasn't in the greylist, the 2nd or 3rd trap _might_ be.

So that got me thinking about lines 353-363 that blacklist boxes if the first trap's type or difficulty meet criteria. I'm wondering if we should also move lines 353-363 to be within the same loop as where you're putting the greylist handling, too.

Thoughts?

![image](https://user-images.githubusercontent.com/68095633/148638262-f20fd529-7de5-4f31-93ea-7d3d722fd47c.png)

Reply:
To avoid the hard-coded magic number `18`, I'm proposing that we either use the `@disarm_careful_threshold` variable which the method `disarm_trap` uses to know to disarm careful, or use a sufficiently large symbolic number that has the connotation of just "yep, this is big and important" like `999` or `Float::Infinity`

```suggestion
          if current_box['trap'] && @trap_greylist.include?(current_box['trap'])
            # The trap may or may not be too difficulty for your skill
            # but the consequence of failure is higher than usual,
            # let's take it safe just in case.
            current_box['trap_difficulty'] = @disarm_careful_threshold
          end        
```

![image](https://user-images.githubusercontent.com/68095633/148638449-9bc0a6d6-ffb4-4863-8d1f-c86e8db477f8.png)

Reply:
> So that got me thinking about lines 353-363 that blacklist boxes if the first trap's type or difficulty meet criteria. I'm wondering if we should also move lines 353-363 to be within the same loop as where you're putting the greylist handling, too, since if the 1st trap doesn't meet blacklist criteria, the 2nd or 3rd trap _might_.

I believe this section is only identifying the speed for this one action and another method (disarm_trap) is actually doing the disarming and making a new determination based on the results of that method.   To me, its easier to see/read in separate sections the way it was done originally. I only want to set it to careful once that box has clearly been "approved" by nesting this check within the "yes, lets disarm this box" and it's most obvious by leaving the too hard sections as they are.

>To avoid the hard-coded magic number 18, I'm proposing that we either use the @disarm_careful_threshold variable which the method disarm_trap uses to know to disarm careful, or use a sufficiently large symbolic number that has the connotation of just "yep, this is big and important" like 999 or Float::Infinity

Setting it to @disarm_careful_threshold probably makes the most sense.  Future proofing. :)

Reply:
Gotcha, thanks